src/main/java/org/bookkeepervisualmanager/bookkeeper/BookkeeperManager.java

```diff
@@ -52,34 +66,57 @@
 
     private static final Logger LOG = Logger.getLogger(BookkeeperManager.class.getName());
 
-    public static final String BK_LEDGERS_PATH = System.getProperty("bk.ledgers.path", "/ledgers");
-    public static final String ZK_SERVER = System.getProperty("zk.servers", "127.0.0.1:2181");
-    public static final int ZK_TIMEOUT = 1000;
-
+    private final ConfigurationStore configStore;
     private final ClientConfiguration conf;
 
     private ZooKeeper zkClient;
     private BookKeeper bkClient;
     private BookKeeperAdmin bkAdmin;
 
-    public BookkeeperManager(String zkServers) throws BookkeeperException {
+    public BookkeeperManager(ConfigurationStore configStore) throws BookkeeperException {
         try {
+            this.configStore = configStore;
+
+            String zkServers = this.configStore.getProperty(PROPERTY_ZOOKEEPER_SERVER,
+                    PROPERTY_ZOOKEEPER_SERVER_DEFAULT);
+            String zkServiceMetadataUri = this.configStore.getProperty(PROPERTY_BOOKKEEPER_METADATA_SERVICE_URI,
+                    PROPERTY_BOKKEEPER_METADATA_SERVICE_URI_DEFAULT);
+
+            int zkSessionTimeout = ConfigurationStoreUtils.getInt(PROPERTY_ZOOKEEPER_SESSION_TIMEOUT,
+                    PROPERTY_ZOOKEEPER_SESSION_TIMEOUT_DEFAULT, this.configStore);
+            int zkFirstConnectionTimeout = ConfigurationStoreUtils.getInt(PROPERTY_ZOOKEEPER_CONNECTION_TIMEOUT,
+                    PROPERTY_ZOOKEEPER_CONNECTION_TIMEOUT_DEFAULT, this.configStore);
+
             this.conf = new ClientConfiguration();
-            this.conf.setMetadataServiceUri("zk+null://" + zkServers.replace(",", ";") + BK_LEDGERS_PATH);
+            this.conf.setMetadataServiceUri(zkServiceMetadataUri);
 
-            LOG.log(Level.INFO, "Starting zookeeper connection with connection string = {0}", zkServers);
+            LOG.log(Level.INFO, "Starting Zookeeper first connection with connection string = {0}", zkServiceMetadataUri);
+            AtomicBoolean zkConnected = new AtomicBoolean(false);
             CountDownLatch countDownLatch = new CountDownLatch(1);
-            this.zkClient = new ZooKeeper(zkServers, ZK_TIMEOUT, e -> {
+            this.zkClient = new ZooKeeper(zkServers, zkSessionTimeout, e -> {
                 switch (e.getState()) {
                     case SyncConnected:
                         LOG.log(Level.INFO, "Zookeeper connection established.");
+                        zkConnected.set(true);
+                        countDownLatch.countDown();
+                        break;
+                    case Disconnected:
+                        LOG.log(Level.INFO, "Zookeeper connection lost.");
+                        break;
+                    case AuthFailed:
+                        LOG.log(Level.INFO, "Zookeeper auth failed.");
+                        zkConnected.set(true);
```

Comment:
please add ad comment, we are not "connected", but it does not make sense to wait anymore, the error is fatal